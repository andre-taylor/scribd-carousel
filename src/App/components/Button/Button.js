import React from 'react';
import PropTypes from 'prop-types';

import styles from './Button.module.css';

export const Button = ({ a11y, children, type }) => {
  return (
    <button type={type}>
      {a11y && <span className={styles.a11y}>{a11y}</span>}
      {children}
    </button>
  );
};

Button.defaultProps = {
  type: 'button',
};

Button.propTypes = {
  a11y: PropTypes.string,
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf(['button']),
};

export default Button;
