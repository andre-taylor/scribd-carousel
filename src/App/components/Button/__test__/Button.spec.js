import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

describe('<Button />', () => {
  it('renders correctly with all optional and required props', () => {
    const component = renderer.create(
      <Button a11y={'Accisbility Text'} type="button">
        <span>"props.children"</span>
      </Button>
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly using defaultProp and required propss', () => {
    const component = renderer.create(
      <Button>
        <span>"props.children"</span>
      </Button>
    );
    expect(component).toMatchSnapshot();
  });
});
