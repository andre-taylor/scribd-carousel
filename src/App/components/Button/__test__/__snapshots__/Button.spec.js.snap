// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly using defaultProp and required propss 1`] = `
<button
  type="button"
>
  <span>
    "props.children"
  </span>
</button>
`;

exports[`<Button /> renders correctly with all optional and required props 1`] = `
<button
  type="button"
>
  <span
    className="a11y"
  >
    Accisbility Text
  </span>
  <span>
    "props.children"
  </span>
</button>
`;
